gems/sorbet-runtime/lib/types/props/utils.rb

```diff
@@ -56,4 +58,17 @@ def self.optional_prop?(prop_rules)
   def self.merge_serialized_optional_rule(prop_rules)
     {'_tnilable' => true}.merge(prop_rules.merge('_tnilable' => true))
   end
+
+  @scalar_modules = T.let({}, T::Hash[Module, T::Boolean])
```

Comment:
Using WeakRef here might be premature optimization/complication; happy to rip it out if reviewer prefers.

Reply:
Let's rip it off. I have a lot of FUD about it comming from JVM where it has funky effects on GC.